From: SeongJae Park <sj@kernel.org>
Date: Mon, 8 Sep 2025 18:32:31 -0700
Subject: [PATCH] ==== kunut improvements ====

mm/damon/tests: add more tests for online parameters commit

A DAMON feature called parameters "commit" allows DAMON API callers and
ABI users to update nearly every DAMON parameter while DAMON is running.
This is being used for flexible DAMON use cases such as taking a
snapshot of the monitoring results with minimum overhead, or adjusting
access-aware system operations (DAMOS) for user-space driven auto-tuning
or investigations.

Compared to the usefulness of the feature and size of the
implementation, the test coverage is pretty small.  Only the filter
commit part has a single test case, namely damos_test_commit_filter().
Actually, we found and fixed a few bugs of the feature in the past.  The
single existing test was also added to avoid reintroduction of a found
bug.

Add more unit tests for the feature.

First four patches (1-4) refactor and extend the existing test for DAMOS
filter commit for multiple test cases.

Next three patches (5-7) add tests for DAMOS quota commit.

Next two patches (8 and 9) refactor damos_commit_dests() for ease of
code reading and test writing, and implement a new unit test of the
function that is being refactored in a test-friendly way.

Final two patches (10 and 11) further add new unit tests for
damos_commit() and damon_commit_target_regions().

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_jol40ka9 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_jol40ka9

diff --git a/hkml_cv_bogus/hkml_cv_bogus_jol40ka9 b/hkml_cv_bogus/hkml_cv_bogus_jol40ka9
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.47.3

